### DISCUSSION TITLE
How to apply user's own 3D images to porespy, rather than images generated?

### DISCUSSION THREAD

**Discussion by eromanga233** [PMEAL/porespy#738] [Q&A] [answered]
In tutorials, images used are totally generated from the generator module. While in true scientific study, researchers tend to use their own images. How can I apply my own images to porespy?

**Comment by jgostick** [comment, marked as answer]
No problem:

``` python
import imageio

im = imageio.volread(<your filename>)
```

If you don't have `imageio` installed, do `pip install imageio` at the conda prompt.  

Your file can be in multiple formats, I like tif since it's common and recognizable as an image.  You also need to ensure your image is in binary format, ie 1's and 0's.  You can do this in python after you've used `imageio` to load it, or preprocess it in something like `imagej`. 

I'll make a note to create a tutorial on this.
